### DISCUSSION TITLE
Conflict between automerge branch and stabilitydays on bitbucket server

### DISCUSSION THREAD

**Discussion by ryden54** [renovatebot/renovate#14469] [Request Help]
### How are you running Renovate?
Renovate in docker on private infra

**If you're self-hosting Renovate, tell us what version of Renovate you run.**
renovate/renovate:31.77

**Please select which platform you are using if self-hosting.**
Debian

**If you're self-hosting Renovate, tell us what version of the platform you run.**
Buster 10.10

### Was this something which used to work for you, and then stopped?
I never saw this working

### Describe the bug
When using "automerge" with "automergeType: branch" with "stabilityDays: 60" (and without ignoreTests), with a repository hosted on a bitbucket server
We expect that for a new version of a dependency:
- a branch is created by renovate with the dep update, with the stability days "build status" associated to it
- the app's CI build plan is triggered, if it exists
- on renovate's next run, if the stability days is ok **and at least one build plan exists** and all build plans are green, the branch is merged by renovate

What actually happens is that, if no CI build plan exists (or is triggered) for this repository, renovate seems to consider the stability day "build result" it associated to the commit, as a successfull build plan, and therefore directly merges the branch

### Is there something that can be done on the projects side to help renovate parse these correctly?
in https://github.com/renovatebot/renovate/blob/5723be1ac5c776cff5c549457bd3cc238c54b5cf/lib/platform/bitbucket-server/index.ts#L418-L420
if there's a stabilityDays config enabled for this branch, renovate could check if there's more than 1 (instead of zero) successfull commitStatuses

One other solution would be to scan all commit statuses and make sure that either ignoreTests is enabled or there's at least one commitStatus that is not renovate's stability days

### Relevant debug logs
None

### Have you created a minimal reproduction repository?
No, but it makes sense reading renovate's code

**Comment by rarkins** [comment]
https://github.com/renovatebot/renovate/issues/7800

**Comment by ryden54** [comment]
Thank you, that's indeed a better description of this issue

to be able to perform this check on bitbucket server, it might be easyer to use https://docs.atlassian.com/bitbucket-server/rest/6.0.0/bitbucket-build-rest.html#idp8 instead of https://docs.atlassian.com/bitbucket-server/rest/6.0.0/bitbucket-build-rest.html#idp5
as it sends back the details of each build result associated to the commit, allowing to filter renovate/* builds
the implementation would therefore be the same as for bitbucket

Until the blocking #12269 is completed, we will probably disabled the stabilityDays feature
If it still need some time, maybe completing/merging #7875 would be profitable?
